packages/playground/src/components/app-root/app-root.tsx

```diff
@@ -219,100 +219,121 @@ export class AppRoot {
       loggedUser.token as string
     );
 
-    this.updateAccount({ ...this.accountState, user: loggedUser });
+    this.updateAccount({ user: loggedUser });
 
     return loggedUser;
   }
 
-  async getBalances() {
-    const { user } = this.accountState;
+  async getBalances(): Promise<{
+    ethFreeBalanceWei: BigNumber;
+    ethMultisigBalance: BigNumber;
+  }> {
+    const {
+      user: { multisigAddress, ethAddress }
+    } = this.accountState;
     const { provider } = this.walletState;
+    const node = CounterfactualNode.getInstance();
 
-    if (!user.multisigAddress || !user.ethAddress) {
-      return;
+    if (!multisigAddress || !ethAddress) {
+      return {
+        ethFreeBalanceWei: ethers.constants.Zero,
+        ethMultisigBalance: ethers.constants.Zero
+      };
     }
 
-    const multisigBalance = parseFloat(
-      ethers.utils.formatEther(
-        (await provider.getBalance(user.multisigAddress)).toString()
-      )
-    );
+    const query = {
+      type: Node.MethodName.GET_MY_FREE_BALANCE_FOR_STATE,
+      requestId: window["uuid"](),
+      params: { multisigAddress } as Node.GetMyFreeBalanceForStateParams
+    };
 
-    const walletBalance = parseFloat(
-      ethers.utils.formatEther(
-        (await provider.getBalance(user.ethAddress)).toString()
-      )
-    );
+    const { result } = await node.call(query.type, query);
 
-    this.updateAccount({
-      ...this.accountState,
-      balance: multisigBalance,
-      accountBalance: walletBalance
-    });
+    const { balance } = result as Node.GetMyFreeBalanceForStateResult;
 
-    return {
-      balance: multisigBalance,
-      accountBalance: walletBalance
+    const vals = {
+      ethFreeBalanceWei: ethers.utils.bigNumberify(balance),
+      ethMultisigBalance: await provider!.getBalance(multisigAddress)
     };
+
+    this.accountState.updateAccount!(vals);
+
+    return vals;
   }
 
-  async deposit(value: string, multisigAddress: string) {
-    const { accountBalance } = this.accountState;
-    const valueInWei = parseInt(value, 10);
+  async deposit(valueInWei: BigNumber) {
+    const {
+      user: { multisigAddress }
+    } = this.accountState;
+
     const node = CounterfactualNode.getInstance();
 
-    this.updateAccount({
-      ...this.accountState,
-      accountBalance: (accountBalance as number) - valueInWei,
-      unconfirmedBalance: (this.accountState.balance as number) + valueInWei
+    node.once(Node.EventName.DEPOSIT_STARTED, args => {
+      this.updateAccount({
+        ethPendingDepositTxHash: args.txHash,
+        ethPendingDepositAmountWei: valueInWei
+      });
     });
 
+    const resetPendingDepositState = () =>
```

Comment:
@snario This shouldn't be an inline function, since it'll be re-declared every time we deposit. Can we expose it as a class function, instead?